server/src/test/scala/org/http4s/server/middleware/DefaultHeadSpec.scala

```diff
@@ -0,0 +1,50 @@
+package org.http4s
+package server
+package middleware
+
+import scalaz._, Scalaz._
+import scalaz.concurrent.Task
+import scalaz.stream.Process._
+import org.http4s.Uri.uri
+import org.http4s.dsl._
+
+class DefaultHeadSpec extends Http4sSpec {
+  val service = DefaultHead(HttpService {
+    case req @ GET -> Root / "hello" =>
+      Ok("hello")
+
+    case GET -> Root / "special" =>
+      Ok().putHeaders(Header("X-Handled-By", "GET"))
+
+    case HEAD -> Root / "special" =>
+      Ok().putHeaders(Header("X-Handled-By", "HEAD"))
+  })
+
+  "DefaultHead" should {
+    "honor HEAD routes" in {
+      val req = Request(Method.HEAD, uri = uri("/special"))
+      service(req).map(_.headers.get("X-Handled-By".ci).map(_.value)) must returnValue(Some("HEAD"))
+    }
+
+    "return truncated body of corresponding GET on fallthrough" in {
```

Comment:
Could we check the `Content-Length` header as well? For that matter, we could check that the `Response` objects are the exact same except the `body`.